File: src/main/java/itmo/is/project/dto/module/storage/StorageModuleBlueprintDto.java
package itmo.is.project.dto.module.storage;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import itmo.is.project.dto.module.BuildCostDto;

public record StorageModuleBlueprintDto(

        @Schema(example = "1")
        @JsonProperty(value = "id", required = true)
        Integer id,

        @Schema(example = "Storage S")
        @JsonProperty(value = "name", required = true)
        String name,

        @JsonProperty(value = "buildCost", required = true)
        BuildCostDto buildCost,

        @Schema(example = "25000")
        @JsonProperty(value = "capacity", required = true)
        Integer capacity
) {
}

File: src/main/kotlin/itmo/isproject/dto/module/storage/StorageModuleBlueprintDto.kt
package itmo.isproject.dto.module.storage

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema
import itmo.isproject.dto.module.BuildCostDto

data class StorageModuleBlueprintDto(

    @Schema(example = "1")
    @JsonProperty(value = "id", required = true)
    val id: Int?,

    @Schema(example = "Storage S")
    @JsonProperty(value = "name", required = true)
    val name: String,

    @JsonProperty(value = "buildCost", required = true)
    val buildCost: BuildCostDto,

    @Schema(example = "25000")
    @JsonProperty(value = "capacity", required = true)
    val capacity: Int?
)
